packages/adapter-d1/src/d1.ts

```diff
@@ -58,73 +51,42 @@ class D1Queryable<ClientT extends StdClient> implements Queryable {
+    const columnTypes = Object.values(getColumnTypes(columnNames, results))
+    const rows = ioResult.results.map((value) => mapRow(value as Object, columnTypes))
 
     return {
       columnNames,
-      // Note: without Object.values the array looks like
-      // columnTypes: [ id: 128 ],
-      // and errors with:
-      // ✘ [ERROR] A hanging Promise was canceled. This happens when the worker runtime is waiting for a Promise from JavaScript to resolve, but has detected that the Promise cannot possibly ever resolve because all code and events related to the Promise's I/O context have already finished.
-      columnTypes: Object.values(columnTypes),
+      // * Note: without Object.values the array looks like
+      // * columnTypes: [ id: 128 ],
+      // * and errors with:
+      // * ✘ [ERROR] A hanging Promise was canceled. This happens when the worker runtime is waiting for a Promise from JavaScript to resolve, but has detected that the Promise cannot possibly ever resolve because all code and events related to the Promise's I/O context have already finished.
+      columnTypes,
       rows,
     }
   }
 
-  private mapD1ToRows(results: any) {
-    const rows: unknown[][] = []
-    for (const row of results) {
-      const entry = Object.keys(row).map((k) => row[k])
-      rows.push(entry)
-    }
-    return rows
-  }
-
   /**
    * Execute a query given as SQL, interpolating the given parameters and
    * returning the number of affected rows.
    * Note: Queryable expects a u64, but napi.rs only supports u32.
    */
   async executeRaw(query: Query): Promise<Result<number>> {
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
     const tag = '[js::execute_raw]'
-    // console.debug(`${tag} %O`, query)
+    debug(`${tag} %O`, query)
 
     return (await this.performIO(query)).map(({ meta }) => meta.rows_written ?? 0)
   }
 
   private async performIO(query: Query): Promise<Result<PerformIOResult>> {
-    // console.debug({ query })
-
     try {
-      // Hack for booleans, we must convert them to 0/1.
-      // ✘ [ERROR] Error in performIO: Error: D1_TYPE_ERROR: Type 'boolean' not supported for value 'true'
-      query.args = query.args.map((arg) => {
-        if (arg === true) {
-          return 1
-        } else if (arg === false) {
-          return 0
-        }
-        // Temporary unblock for "D1_TYPE_ERROR: Type 'bigint' not supported for value '20'"
-        // For 0-legacy-ports.query-raw tests
-        // https://github.com/prisma/team-orm/issues/878
-        else if (typeof arg === 'bigint') {
-          return Number(arg)
-        } else if (arg instanceof Uint8Array) {
-          return Array.from(arg)
-        }
-
-        return arg
-      })
+      query.args = query.args.map((arg) => this.cleanArg(arg))
 
       const result = await this.client
         .prepare(query.sql)
         .bind(...query.args)
+        // TODO use .raw({ columnNames: true }) later
         .all()
```

Comment:
We need to wait for a release from their side, though it's in the runtime already in the Cloud
and it will be best in a separate PR